baseapp/baseapp.go

```diff
@@ -303,20 +301,13 @@ func (app *BaseApp) init() error {
 	app.setCheckState(tmproto.Header{})
 	app.Seal()
 
-	// make sure the snapshot interval is a multiple of the pruning KeepEvery interval
-	if app.snapshotManager != nil && app.snapshotInterval > 0 {
-		rms, ok := app.cms.(*rootmulti.Store)
-		if !ok {
-			return errors.New("state sync snapshots require a rootmulti store")
-		}
-		pruningOpts := rms.GetPruning()
-		if pruningOpts.KeepEvery > 0 && app.snapshotInterval%pruningOpts.KeepEvery != 0 {
-			return fmt.Errorf(
-				"state sync snapshot interval %v must be a multiple of pruning keep every interval %v",
-				app.snapshotInterval, pruningOpts.KeepEvery)
-		}
+	rms, ok := app.cms.(*rootmulti.Store)
+	if !ok {
+		return errors.New("state sync snapshots require a rootmulti store")
+	}
+	if err := rms.GetPruning().Validate(); err != nil {
+		return err
 	}
```

Comment:
shouldn't this still be inside of an `app.snapshotManager != nil`?

Reply:
No because pruning options are set independently from snapshot settings. If we do not explicitly set pruning options, pruning-nothing is set by default.

Pruning can and should work independently from snapshots and vice versa.

However, if snapshot are set, then it supplies `snapshot-interval` to the `pruning.Manager`. As a result, `pruning.Manager` knows which heights to skip until after the snapshot is taken.

Let me know if that makes sense

Reply:
I guess I'm a bit thrown off that app.cms has to be a root multi store here. Perhaps the logic should be:

```suggestion
	rms, ok := app.cms.(*rootmulti.Store)
	if !ok && app.snapshotInterval > 0 {
		return errors.New("state sync snapshots require a rootmulti store")
	} else if ok {
	  if err := rms.GetPruning().Validate(); err != nil {
		  return err
	  }
	}
```

Reply:
That's correct, there should have been a check similar to the suggested:
```
if !ok && app.snapshotManager != nil {
		return errors.New("state sync snapshots require a rootmulti store")
}
if err := rms.GetPruning().Validate(); err != nil {
                return err
}
```

Updated. Thanks for catching that